### PR TITLE
Release/setup4

### PR DESCRIPTION
**Pull request by aradradev**
In this pull request, I implemented the following features:

- Make use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as the backend.
- Used [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make the HTTP requests to the backend
- Fetched the books from the Bookstore API after the component has rendered
- Refactored the add book and remove book features to persist the changes in the server

Please review this pull request any feedback for improvement is highly appreciated.

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  async (item_id) => {
+    try {
+      await axios.delete(`${url}/${item_id}`);
+      return item_id;
+    } catch (error) {
+      return error;
+    }
+  },
+);
+
```
Comment (DammyShittu) [inline src/redux/books/booksSlice.js]: - [ ] The book add and fetch are well implemented. Books can be stored on the API and reloading the browser fetches the book. But the delete functionality does not really work. Whenever I click the delete button for just one book, the page goes blank as if all the books are removed (kindly fix this also). But when I refresh the page all the books are retrieved and displayed. Kindly fix that.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @aradradev,

## Highlights
✔️ Add and fetch books features

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (aradradev) [reply src/redux/books/booksSlice.js]: @DammyShittu Please I am still struggling to remove one specific item book, I tried with one of my standUP team but nothing. Can you give some hints or any resources if you have time?

**Review by julie-ify — APPROVED** [review]
Hi @aradradev,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**